js/src/structured-data-blocks/faq/components/FAQ.js

```diff
@@ -0,0 +1,336 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Question from "./Question";
+import { stripHTML } from "../../../helpers/stringHelpers";
+import isUndefined from "lodash/isUndefined";
+
+const { __ } = window.wp.i18n;
+const { RichText } = window.wp.editor;
+const { IconButton } = window.wp.components;
+const { Component, renderToString } = window.wp.element;
+
+/**
+ * A FAQ block component.
+ */
+export default class FAQ extends Component {
+
+	/**
+	 * Constructs a FAQ editor component.
+	 *
+	 * @param {Object} props This component's properties.
+	 *
+	 * @returns {void}
+	 */
+	constructor( props ) {
+		super( props );
+
+		this.state = { focus: "" };
+
+		this.changeQuestion = this.changeQuestion.bind( this );
+		this.insertQuestion = this.insertQuestion.bind( this );
+		this.removeQuestion = this.removeQuestion.bind( this );
+		this.swapQuestions = this.swapQuestions.bind( this );
+
+		this.setFocus = this.setFocus.bind( this );
+
+		this.editorRefs = {};
+	}
+
+	/**
+	 * Generates a pseudo-unique" id.
+	 *
+	 * @param {string} prefix an (optional) prefix to use.
+	 *
+	 * @returns {string} a pseudo-unique string, consisting of the optional prefix + the curent time in milliseconds.
+	 */
+	static generateId( prefix ) {
+		return `${ prefix }-${ new Date().getTime() }`;
+	}
+
+	/**
+	 * Replaces the FAQ Question with the given index.
+	 *
+	 * @param {array|string} newQuestion  The new contents of the question.
+	 * @param {array|string} newAnswer    The new contents of the answer to this question.
+	 * @param {array}        prevQuestion The old question.
+	 * @param {array}        prevAnswer   The old answer.
+	 * @param {number}       index        The index of the Question that needs to be changed.
```

Comment:
I would make `Question` lowercase, because this method does not have an argument that is called `Question`